src/net.cpp

```diff
@@ -2410,7 +2411,7 @@ int CConnman::GetExtraFullOutboundCount() const
             }
         }
     }
-    return std::max(full_outbound_peers - m_max_outbound_full_relay, 0);
+    return full_outbound_peers - m_max_outbound_full_relay;
```

Comment:
Negative numbers mean neither of those code blocks are executed; if it was `max{...,0}` then the "at zero" code block would be executed when it shouldn't be.

Reply:
Right, because we want to execute code for the case in which we are exactly at the limit, and being under the limit will be mapped to the same value. My bad.

I have to say though that, after this change, the function name becomes a bit confusing.

Reply:
> I have to say though that, after this change, the function name becomes a bit confusing.

I had a quick look at changing it to `return full_outbound_peers` and doing the comparison with `m_max_outbound_full_relay` in the caller. Would require exposing the latter value to net_processing, but otherwise seems like it works. Wasn't convinced it was enough of an improvement to bother though.

Reply:
I could add this if reviewers prefer it. I also thought about renaming, but couldn't really think of a better name than "GetExtraFullOutboundCount" that would imply negative return values.

Reply:
I like that name! I missed your comment in the most recent push, but will change it tomorrow!

Reply:
done now!